app/src/main/java/com/mapbox/maps/testapp/examples/style/TileJsonActivity.kt

```diff
@@ -65,6 +69,38 @@ class TileJsonActivity : AppCompatActivity() {
     }
   }
 
+  override fun onCreateOptionsMenu(menu: Menu): Boolean {
+    menuInflater.inflate(R.menu.menu_tilejson, menu)
+    return true
+  }
+
+  override fun onOptionsItemSelected(item: MenuItem): Boolean {
+    when (item.itemId) {
+      R.id.menu_set_tile_request_delay -> {
+        item.isChecked = !item.isChecked
+        setTileDelay(TILE_REQUEST, item.isChecked)
+      }
+      R.id.menu_set_tile_network_request_delay -> {
+        item.isChecked = !item.isChecked
+        setTileDelay(NETWORK_REQUEST, item.isChecked)
+      }
+    }
+    return true
+  }
+
+  private fun setTileDelay(requestType: String, isChecked: Boolean = false) {
+    mapboxMap.getStyle {
+      it.getSourceAs<RasterSource>(SOURCE_ID).apply {
+        if (requestType == TILE_REQUEST) {
+          this?.tileRequestsDelay(if (isChecked) RASTER_TILE_DELAY else DEFAULT_RASTER_TILE_DELAY)
```

Comment:
feels a bit strange to me to read or write `this?`

Reply:
Can change to `it.getSourceAs<RasterSource>(SOURCE_ID)?.apply `